tests/unit/test_req_file.py

```diff
@@ -301,6 +301,30 @@ def test_extras_for_editable_url_requirement(self):
         assert req.extras[0] == 'ex1'
         assert req.extras[1] == 'ex2'
 
+    def test_options_and_markers(self):
+        line = 'SomeProject ; python_version=="2.6" --install-option "yo"'
```

Comment:
So this is fine too, I have no ego in which approach is taken.

Can I suggest though that if the intent is that options only be allowed to follow args, that we need to include a INCOMPATIBLE warning in 7.0.2 because 7.0.0 and 7.0.1 supported this:

--install-options "foo" requirementnamehere

as a side effect of the introduction of optparse parsing. Which is why I took the more complex approach I did of lifting out the marker section - so I wouldn't introduce a further incompatibility.